node/pkg/watchers/solana/client.go

```diff
@@ -1183,3 +1161,18 @@ func (s *SolanaWatcher) checkCommitment(commitment rpc.CommitmentType, isReobser
 	}
 	return true
 }
+
+// isPossibleWormholeMessage searches the logs on a transaction to see if it contains a Wormhole core PostMessage.
+// It looks for a log for the core contract, followed by one for the sequence number.
+func isPossibleWormholeMessage(whLogPrefix string, logMessages []string) bool {
+	for idx := 0; idx < len(logMessages); idx++ {
+		if strings.HasPrefix(logMessages[idx], whLogPrefix) {
+			for idx1 := idx + 1; idx1 < len(logMessages); idx1++ {
+				if strings.HasPrefix(logMessages[idx1], "Program log: Sequence:") {
+					return true
+				}
```

Comment:
since this loops over the remainder of the list, you can add an `else { return false }` here to end earlier (as is, this would go back and continue looping over the rest of the log messages again)

Reply:
meant to put this comment two lines down :grimacing: